src/main/java/org/zalando/nakadi/security/ScopeHelper.java

```diff
@@ -0,0 +1,25 @@
+package org.zalando.nakadi.security;
+
+import org.zalando.nakadi.exceptions.IllegalScopeException;
+
+import java.util.Objects;
+import java.util.Set;
+
+public class ScopeHelper {
+
+    public static void checkScopes(final Set<String> allowedScopes, final Set<String> clientScopes) {
+        if (isEmptyScope(allowedScopes) ) {
+            return;
+        }
+
+        allowedScopes.stream()
+                .filter(clientScopes::contains)
+                .findAny()
+                .orElseThrow(() -> new IllegalScopeException(allowedScopes));
+    }
+
+    private static boolean isEmptyScope(final Set<String> allowedScopes) {
+        return Objects.isNull(allowedScopes) || allowedScopes.isEmpty();
```

Comment:
1. I see nothing strange here, I think it is up to developer how to check it. Someone loves Joda style, others not and same here for this check :)